js/main.js

```diff
@@ -10,35 +10,48 @@ var $modal = document.querySelector('.modal-overlay');
 var $closeModal = document.querySelector('#close-modal');
 var $bookmarkButtonNav = document.querySelector('#nav-bookmarks');
 var $bookmarkButtonModal = document.querySelector('.go-to-bookmark');
+var $confirmMessage = document.querySelector('.confirm-message');
+var $bookmarkMessage = document.querySelector('.bookmark-message');
 
 $searchForm.addEventListener('submit', handleSubmit);
 $searchButton.addEventListener('click', dataView);
 $navSearchButton.addEventListener('click', dataView);
 window.addEventListener('DOMContentLoaded', handleLoad);
 $ul.addEventListener('click', handleDetail);
-$ul.addEventListener('click', handleAdd);
+$ul.addEventListener('click', handleAddAndRemove);
 $back.addEventListener('click', dataView);
 $closeModal.addEventListener('click', closeModal);
 $bookmarkButtonNav.addEventListener('click', handleBookmarks);
 $bookmarkButtonModal.addEventListener('click', handleBookmarks);
+$bookmarkButtonModal.addEventListener('click', confirmRemove);
+$detailPageContainer.addEventListener('click', handleAddAndRemove);
 
 function handleLoad(event) {
   viewSwap(data.view);
+  data.nextResultId = 0;
```

Comment:
What is the purpose of this line? It looks like you are resetting your ids every time the page loads. That means there is a high likelihood you will accidentally make multiple items with the same id at some point.